tests/test_main.py

```diff
@@ -729,6 +743,10 @@ def test_sample_files(self, full_filename, result_filename):
                 assert result == json.load(infile)
 
 
+    @pytest.mark.skipif(
+        NodeFinder.__name__ == "SentinelNodeFinder",
```

Comment:
Should this still be here?

Reply:
yes, I tried to fix the problems in the SentinelNodeFinder in #74, but there are still open issues.

The general problem with `test_module_files` is that the tests in ci are not deterministic. I have really good experiences with `tests/generate_small_sample.py`, which I used to collect problematic source code pieces from a larger set of source files (this is what I started to do in #74). 

I would recommend to continue on #74 or to use #64 for cpython >= 3.8

Reply:
I would like to delete this test later and build something which generates small edge cases ... see https://github.com/15r10nk/pysource-codegen. I found several cpython 3.12 bugs with this generated code and would like to integrate it also in the executing tests.

Reply:
Coveralls says that coverage decreased by 2.9%, and I'm wondering if that's because of this. Please reenable the test so we can see what happens?

Reply:
OK, the test is passing and coverage has only decreased 0.4%.

Reply:
Great, I will try to generate small_samples which will cover the uncovered lines when I have some time.
Generating code which increases coverage sounds interesting. Should be quite similar to the mutmut stuff I used to find samples for some edge cases.

Does deprecating 2.7 require a new major release?

Reply:
Yes, let's do a major release